packages/pyright-internal/src/analyzer/importStatementUtils.ts

```diff
@@ -160,10 +160,16 @@ export function getTextEditsForAutoImportSymbolAddition(
             // Insert new symbol by import symbol type and then alphabetical order.
             // Match isort default behavior.
             const symbolNameType = _getImportSymbolNameType(symbolName);
+            // isort will prefer '_' over alphanumerical chars
+            // This can't be reproduced by a normal string compare in TypeScript, since '_' > 'A'.
+            // Replace all '_' with '=' which guarantees '=' < 'A'.
+            // Safe to do as '=' is an invalid char in Python names.
+            const symbolNameCompare = symbolName.replace('_', '=');
```

Comment:
```suggestion
            const symbolNameCompare = symbolName.replace(/_/g, '=');
```

I'm thinking you intended this to replace all underscores? replace only replaces the first match unless you use a regex. (We can't use replaceAll yet, so this is the right way to replace all.)

Reply:
👍🏻 Updated the code and added an additional test case for it.

Reply:
Good catch @jakebailey!